src/pages/in/embed/nft/class/_classId.vue

```diff
@@ -0,0 +1,190 @@
+    :description="contentDescription"
+    :img-src="contentImage"
+    :url="contentURL"
+    :price="nftPrice"
+    :collector-count="nftCollectorCount"
+    :collected-count="nftCollectedCount"
+    :owner-address="iscnOwnerAddress"
+    :owner-display-name="iscnOwnerDisplayName"
+    :owner-avatar-src="iscnOwnerAvatarSrc"
+    :is-content-clickable="false"
+    @view-details="viewNFTDetails"
+    @collect="collectNFT"
+    @like="likeISCN"
+    @load-image="notifyParentOfResizing"
+  />
+</template>
+
+<script>
+import {
+  APP_LIKE_CO_URL_BASE,
+  LIKER_LAND_URL_BASE,
+} from '~/constant';
+import {
+  apiGetLikerDataByAddress,
+  apiGetNFTMetadata,
+  apiGetNFTOwners,
+  apiGetNFTPurchaseInfo,
+} from '~/util/api/api';
+import { getClassInfo } from '~/util/nft';
+
+export default {
+  layout: 'widget',
+  computed: {
+    widgetId() {
+      return this.$route.query.wid;
+    },
+    detailsURL() {
+      return `${LIKER_LAND_URL_BASE}/nft/class/${this.classId}`;
+    },
+    purchaseURL() {
+      return `${APP_LIKE_CO_URL_BASE}/nft/purchase/${encodeURIComponent(
+        this.iscnId
+      )}`;
+    },
+  },
+  async asyncData({ params }) {
+    const { classId } = params;
+    let metadata;
+    const [
+      chainMetadata = {},
+      apiMetadataResult,
+      ownersResult,
+      purchaseInfoResult,
+    ] = await Promise.all([
+      getClassInfo(classId),
+      apiGetNFTMetadata({ classId })
+        // eslint-disable-next-line no-console
+        .catch(error => console.error(error)),
+      apiGetNFTOwners({ classId })
+        // eslint-disable-next-line no-console
+        .catch(error => console.error(error)),
+      apiGetNFTPurchaseInfo({ classId })
+        // eslint-disable-next-line no-console
+        .catch(error => console.error(error))
+    ]);
+    const {
+      name,
+      description,
+      data: {
+        metadata: classMetadata = {},
+        parent,
+      } = {},
+    } = chainMetadata;
+    metadata = {
+      name,
+      description,
+      metadata: classMetadata,
+      parent,
+    };
+    if (apiMetadataResult && apiMetadataResult.data) {
+      metadata = {
+        ...metadata,
+        ...apiMetadataResult.data,
+      };
+    }
+    const {
+      name: contentTitle,
+      description: contentDescription,
+      image: contentImage,
+      metadata: {
+        url: contentURL,
+      } = {},
+      iscn_id: iscnId,
+      iscn_owner: iscnOwnerAddress,
+    } = metadata;
+    const {
+      price: nftPrice,
+    } = (purchaseInfoResult || {}).data || {};
+    const ownersMap = (ownersResult || {}).data || {};
+    const nftCollectedCount = Object.values(ownersMap).reduce(
+      (count, collected) => count + collected.length,
+      0
+    );
+    const nftCollectorCount = Object.keys(ownersMap).length;
+
+    const likerDataResult = await apiGetLikerDataByAddress(iscnOwnerAddress)
+      // eslint-disable-next-line no-console
+      .catch(error => console.error(error));
```

Comment:
You can save a `|| {}` guard below if you return `{}` here, not sure which one is cleaner